lib/jekyll/tags/highlight.rb

```diff
@@ -81,20 +81,32 @@ def render_pygments(code, _context)
       def render_rouge(code)
         require "rouge"
         formatter = ::Rouge::Formatters::HTML.new
-        if @highlight_options[:linenos]
-          formatter = ::Rouge::Formatters::HTMLTable.new(
-            formatter,
-            {
-              :css_class    => "highlight",
-              :gutter_class => "gutter",
-              :code_class   => "code",
-            }
-          )
-        end
+        formatter = line_highlighter_formatter(formatter) if @highlight_options[:highlight_lines]
+        formatter = table_formatter(formatter) if @highlight_options[:linenos]
+
         lexer = ::Rouge::Lexer.find_fancy(@lang, code) || Rouge::Lexers::PlainText
         formatter.format(lexer.lex(code))
       end
 
+      def line_highlighter_formatter(formatter)
+        ::Rouge::Formatters::HTMLLineHighlighter.new(
+          formatter,
+          :highlight_lines      => @highlight_options[:highlight_lines].map(&:to_i),
+          :highlight_line_class => @highlight_options[:highlight_line_class] || "hll"
```

Comment:
I don't see a test for `highlight_line_class` in the PR diff... did I miss that? When I read it, I thought it was for selecting a Ruby class within Rouge, but looking at it again I think it's to add a CSS class to... some element in the output. A test would help illuminate this and ensure we don't break it in the future. Can it take a list of classes?

Reply:
@parkr I think [these lines in the Cucumber test](https://github.com/jekyll/jekyll/pull/9138/files#diff-460702aaee67a2a9b221cdee16477ef0fe3202c7eef2f8ee18e9714a3b965543R60-R61) covers the `hll` bit removing the shadow of doubt:
```gherkin
And I should see "<span class=\"hll\"><span class=\"k\">module</span> <span class=\"nn\">Jekyll</span>" in "_site/index.html"
And I should see "<span class=\"hll\">  <span class=\"k\">module</span> <span class=\"nn\">Tags</span>" in "_site/index.html"
```
@lylo Please add **two supersets** of the current scenario with custom classnames (one for valid unquoted input, the other for invalid *quoted* input)

Reply:
@ashmaroli @parkr thanks for picking this up. I'm new to cucumber so I'm not entirely sure what you mean by "two supersets" but I've added new scenarios to cover both the custom line highlighting class, as well as syntax validation. See 97e60d1d8b0b239b1fa9358655dfac92ed270a4c and 56d6d8115a1c8d532a0cd50190ed8775e0b303cc. If there's a more succinct way, please do let me know and I'll be happy to change.

Reply:
This has now been addressed. Apologies for the confusion with the renamed test file there, that was my ignorance - not paying attention.

I've also updated `highlight.rb` so that it throws a syntax error rather than bailing with an obscure (to the user) Ruby error https://github.com/jekyll/jekyll/pull/9138/commits/2dbb1812397cc19165f2bc095f84035cc0c3838e#diff-d38972f992a69af4316684fc12c85a72ca6332c30d73359334eb614fdba826b9R92